Doc(eos_cli_config_gen, eos_designs): Consistent descriptions for BGP AS schema fields re asdot notation

ClausHolbechArista opened the pull request:
## Change Summary

<!-- Enter short PR description -->
- Improve descriptions for BGP AS schema fields re asdot notation

We should not implement convert_types from `float` since it would introduce a risk when the user inputs `65000.100`, we would configure `65000.1`. **We already have that a few places, so to avoid breaking changes, this is kept in this PR.**

## Component(s) name

`arista.avd.eos_designs`
`arista.avd.eos_cli_config_gen`

ClausHolbechArista committed:
Doc(eos_cli_config_gen): Improve descriptions for BGP AS schema fields re asdot notation.

carl-baillargeon reviewed (commented):
The added description is missing under network service: `<network_services_keys.name>.[].vrfs.[].bgp_peers.[].remote_as`.

ClausHolbechArista committed:
Update schemas for eos_designs as well

ClausHolbechArista committed:
Merge branch 'devel' into doc/eos_cli_config_gen/bgp-as-asdot-descriptions

ClausHolbechArista commented:
> The added description is missing under network service: `<network_services_keys.name>.[].vrfs.[].bgp_peers.[].remote_as`.

The scope was only `eos_cli_config_gen`, but I have fixed it for `eos_designs` as well now, and I actually found a few places where we converted from `float`. So since I change that I have changed this to a fix instead of doc.

ClausHolbechArista committed:
Revert convert_type changes

ClausHolbechArista committed:
Merge branch 'devel' into doc/eos_cli_config_gen/bgp-as-asdot-descriptions

carl-baillargeon reviewed (commented):
The `*must*` is not rendered properly. I _think_ it was fine when I first reviewed.

![image](https://github.com/aristanetworks/avd/assets/63206086/07e3e4d5-e980-47a3-be7d-1415b6683aa6)

carlbuchmann commented:
> The `*must*` is not rendered properly. I _think_ it was fine when I first reviewed.
> 
> ![image](https://private-user-images.githubusercontent.com/63206086/305270078-07e3e4d5-e980-47a3-be7d-1415b6683aa6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgwNTM4MDIsIm5iZiI6MTcwODA1MzUwMiwicGF0aCI6Ii82MzIwNjA4Ni8zMDUyNzAwNzgtMDdlM2U0ZDUtZTk4MC00N2EzLWJlN2QtMTQxNWI2NjgzYWE2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE2VDAzMTgyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkzNDQ1OGNhN2UzZmQwZGQwNjAzZDQzYzE3ZWY5YjkyNDE3ODU5ZjJiNzY3NzU2MGU0M2QxYzg3ZWU1ODcxMzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.gXC09dy3HO3wcAP0-qu0TnXM_UjfpRXrjimYk-NvCiE)

That is to be expected in the YAML output. It is italic in the MD table as expected.But perhaps we want bold?

carlbuchmann reviewed (approved):
LGTM, small comment on format

ClausHolbechArista committed:
Update formatting

chetryan reviewed (approved):
LGTM - much clearer descriptions than before.
